### PR TITLE
Fix running documentation make targets on MacOS

### PR DESCRIPTION
**Pull request by chancez**
- `xargs -l` isn't POSIX, so use `-L 1`, 
- Remove `-0` from xargs which didn't seem to be necessary, but was causing issues passing the file names from grep to `sed`.
- Support overriding sed command. MacOS uses BSD sed, which requires an argument for `-i`. Overriding sed allows using GNU sed via `make SED=gsed`.

```release-note
Fix running documentation make targets on MacOS
```

### CODE REVIEW

File: install/kubernetes/Makefile [hunk 1]
```diff
@@ -26,6 +26,7 @@ DEV_VERSION_REGEX := '[0-9]\+\.[0-9]\+-dev'
 CILIUM_CHART_REGEX := '\([vV]ersion:\) '$(VERSION_REGEX)
 CILIUM_PULLPOLICY_REGEX := '\([pP]ullPolicy:\) .*'
 
+SED := sed
```
Comment (sayboras) [inline install/kubernetes/Makefile]: nit: can you move this to Makefile.defs ? so that later we can easily swap sed command in main Makefile as well.

Comment (sayboras) [reply install/kubernetes/Makefile]: and should it be `SED ?= sed` ?

**Review by sayboras — COMMENTED** [review]
I tried to run with default option in macos (e.g. make -C install/kubernetes all), and it creates one temp file like below, so overriding sed with gsed is a must in macos I reckon. Not sure how hard it is to set `sed` to right value based on arch/os in Makefile.

```
$ gst
On branch pr/chancez/generate_docs_macos
Untracked files:
  (use "git add <file>..." to include in what will be committed)
	install/kubernetes/cilium/Chart.yaml-e
```

File: install/kubernetes/Makefile (hunk 1 above)
Comment (chancez) [reply install/kubernetes/Makefile]: Both of those seem reasonable. TBH I never really know when to use different assignment operators, I need to re-read that section on Make.

**Comment by chancez** [comment]
> I tried to run with default option in macos (e.g. make -C install/kubernetes all), and it creates one temp file like below, so overriding sed with gsed is a must in macos I reckon. Not sure how hard it is to set `sed` to right value based on arch/os in Makefile.
> 
> ```
> $ gst
> On branch pr/chancez/generate_docs_macos
> Untracked files:
>   (use "git add <file>..." to include in what will be committed)
> 	install/kubernetes/cilium/Chart.yaml-e
> ```

Yeah, I opted to just make the user specify it. `-i` on BSD sed expects you to supply it a suffix for backup files, so it's taking the `-e` and assuming that's the suffix you want. I know how to check for binaries with bash, but I always have trouble scripting that kind of logic within Makefiles.

**Comment by chancez** [comment]
I just found an example from cilium/ebpf, I can try to auto detect the `sed` command to use based on how it autodetects podman vs docker.

File: install/kubernetes/Makefile (hunk 1 above)
Comment (chancez) [reply install/kubernetes/Makefile]: Updated.

File: Makefile.defs [hunk 2]
```diff
@@ -30,6 +30,10 @@ INSTALL = install
 
 CONTAINER_ENGINE?=docker
 
+# use gsed if avaiable, otherwise use sed.
+# gsed is needed for MacOS to make in-place replacement work correctly.
+SED ?= $(if $(shell command -v gsed), gsed, sed)
+
```
Comment (sayboras) [inline Makefile.defs:36]: :100:

**Review by sayboras — APPROVED** [review]
LGTM :1st_place_medal:

**Comment by sayboras** [comment]
Full regression test is not required, the changes are verified as part of GHA for linux, and locally for macos.

Marking this ready to merge.

**Pull request merged**